File: app/src/main/java/com/cappielloantonio/play/subsonic/models/MediaType.java
package com.cappielloantonio.play.subsonic.models;

public enum MediaType {
    MUSIC("music"),
    PODCAST("podcast"),
    AUDIOBOOK("audiobook"),
    VIDEO("video");

    private final String value;

    MediaType(String v) {
        value = v;
    }

    public String value() {
        return value;
    }

    public static MediaType fromValue(String v) {
        for (MediaType c : MediaType.values()) {
            if (c.value.equals(v)) {
                return c;
            }
        }
        throw new IllegalArgumentException(v);
    }
}

File: app/src/main/java/com/cappielloantonio/play/subsonic/models/MediaType.kt
package com.cappielloantonio.play.subsonic.models;

import com.tickaroo.tikxml.annotation.Attribute;
import com.tickaroo.tikxml.annotation.Xml;

@Xml
public class MediaType {
    public static String MUSIC = "music";
    public static String PODCAST = "podcast";
    public static String AUDIOBOOK = "audiobook";
    public static String VIDEO = "video";

    @Attribute
    private String value;

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}
